aleph-client/src/elections.rs

```diff
@@ -1,20 +1,18 @@
-use pallet_elections::EraValidators;
-use primitives::CommitteeSeats;
+use primitives::SessionIndex;
+pub use primitives::{CommitteeSeats, EraValidators};
 use sp_core::H256;
 use substrate_api_client::AccountId;
 
-use crate::AnyConnection;
+use crate::{get_block_hash, get_session_period, AnyConnectionExt};
 
 const PALLET: &str = "Elections";
 
-pub fn get_committee_seats<C: AnyConnection>(
+pub fn get_committee_seats<C: AnyConnectionExt>(
     connection: &C,
     block_hash: Option<H256>,
 ) -> CommitteeSeats {
     connection
-        .as_connection()
-        .get_storage_value(PALLET, "CommitteeSize", block_hash)
-        .expect("Failed to obtain CommitteeSize extrinsic!")
+        .read_storage_value_from_block(PALLET, "CommitteeSize", block_hash)
         .unwrap_or_else(|| {
             panic!(
                 "Failed to decode CommitteeSize for block hash: {:?}.",
```

Comment:
if you follow my suggestion, this unwrapping would go inside `read_storage_value_from_block`

Reply:
Changed.